jax/_src/lax/control_flow/for_loop.py

```diff
@@ -446,6 +447,136 @@ def _for_partial_eval(trace: pe.JaxprTrace, *tracers: pe.JaxprTracer,
   return merge_lists(in_unknowns, known_outputs, unknown_outputs)
 pe.custom_partial_eval_rules[for_p] = _for_partial_eval
 
+def _for_partial_eval_custom(saveable, in_unknowns, in_inst, eqn):
+  jaxpr, nsteps, reverse, which_linear = split_dict(
+      eqn.params, ["jaxpr", "nsteps", "reverse", "which_linear"])
+  num_inputs = len(eqn.invars)
+  # We first need to run a fixpoint to determine which of the `Ref`s are unknown
+  # after running the for loop. We want to use the jaxpr to determine which
```

Comment:
Are the first and second sentences redundant?

Reply:
Yea, edited.